File: 2025-1/sesion-de-clase-02/c#/Persona.cs
using System;

public class Persona : IDisposable {
	private string nombre;
	private int edad;
	
	private bool disposed = false;
	
	public Persona() {
		Console.WriteLine("El objeto se esta construyendo.");
	}
	
	public void Dispose() {
		Console.WriteLine("El objeto se esta liberando de forma explicita.");
		GC.SuppressFinalize(this);
	}
	
	~Persona() {
		Console.WriteLine("El objeto se esta destruyendo.");
	}
}

File: 2025-1/sesion-de-clase-02/java/Persona.java
public class Persona implements AutoCloseable {
	private String nombre;
	private int edad;
	
	public Persona() {
		System.out.println("Constructor de Persona.");
	}
	
	@Override
	public void close() {
		System.out.println("El objeto se esta destruyendo");
	}
	
	/*public void finalize() {
		System.out.println("El objeto se esta destruyendo");
	}*/
}